eng/scripts/update-pathversions.rs

```diff
@@ -103,38 +139,84 @@ fn get_package_versions(toml_files: &Vec<TomlInfo>) -> Vec<(String, String, bool
+            toml_file.is_publish_disabled,
+        ));
     }
 
     package_versions
 }
 
-fn update_package_versions(toml: &mut Table, package_versions: &Vec<(String, String, bool)>, add: bool) {
+fn update_package_versions(
+    toml: &mut Table,
+    package_versions: &Vec<(String, String, bool)>,
+    add: bool,
+) -> bool {
+    // Returns `true` if any modifications were made to the TOML table, `false` otherwise.
+    // This indicates whether the file needs to be written back to disk.
+    //
     // for each dependency table, for each package in package_versions
     // if the package is in the dependency table
     //   if the dependency has both path and version properties, update the version property
     //   if the dependency has has path, but not version, add the version property only if
     //     1. the table name is not "dev-dependencies"
     //     2. the package is not publish disabled
     //     3. the add flag is true
+    //
+    let crate_name = toml
+        .get("package")
+        .and_then(Item::as_table)
+        .and_then(|table| table.get("name"))
+        .and_then(Item::as_str)
+        .unwrap_or("<workspace>")
+        .trim_matches('"')
+        .trim()
+        .to_string();
 
     let dependency_tables = get_dependency_tables(toml);
-
+    let mut updated = false;
     for (table_name, table) in dependency_tables {
         for (package, version, is_publish_disabled) in package_versions {
             if let Some(dependency) = table.get_mut(package) {
                 // azure_idenentity will only be a transitive dev-dependency
-                let should_add = add && table_name != "dev-dependencies" && !is_publish_disabled && package != "azure_identity";
+                let should_add = add
+                    && table_name != "dev-dependencies"
+                    && !is_publish_disabled
+                    && package != "azure_identity";
 
                 let has_path_property = dependency.get("path").is_some();
                 let has_version_property = dependency.get("version").is_some();
 
-                if has_path_property && (has_version_property || should_add) {
-                    dependency["version"] = value(version);
+                if has_path_property {
+                    if has_version_property {
+                        let current_version = dependency
+                            .get("version")
+                            .and_then(Item::as_str)
+                            .unwrap_or("");
+                        if current_version != version {
+                            dependency["version"] = value(version);
+                            println!(
+                                "set {} to version {} in {} {}",
+                                package, version, crate_name, table_name
+                            );
+                            updated = true;
+                        }
+                    } else if should_add {
```

Comment:
Post-GA, shouldn't we remove the version or is the assumption here that the service crate devs would've already had to do that after `azure_core`, for example, shipped the week before?

Reply:
I think we need to spec out our dependency version scheme.  I can't remember what we've said.

Reply:
> I think we need to spec out our dependency version scheme. I can't remember what we've said.

Hence the old issue, https://github.com/Azure/azure-sdk-for-rust/issues/2475